### DISCUSSION TITLE
Change in UsersService.readByQuery: it used to return a single item, now it returns an array

### DISCUSSION THREAD

**Discussion by gdpdils** [directus/directus#5878] [Archive]
Hi, I just updated my app from Directus `9.0.0-rc.59` to Directus `9.0.0-rc.69` and noticed a change in the behaviour of `UsersService.readByQuery`. Specifically, the following code

```javascript
	const user = await userService.readByQuery({
		fields: ['id', 'email'],
		filter: {
			email: aadUser.userPrincipalName,
			status: 'active',
		},
		single: true,
	});
```

used to return a _single instance_ of `user` while now it returns an _array with one user_ even if `single` is set to `true`. Is this a bug or the expected result and hence the way the api is going to work from now on? If so, is this the new behaviour for all services' `readByQuery` methods? Thank you.

**Comment by rijkvanzanten** [comment]
Single isn't / wasn't an officially supported query param (https://docs.directus.io/reference/api/query/), so that's expected 🙂 I'd recommend replacing it with `limit: 1` and pull `user[0]` 👍🏻
